### PR TITLE
Add support for Qwen3Next model with Liger kernels

### PR DESCRIPTION
**Pull request by vvvdwbvvv**
## Summary

Adding qwen3-next model support  #896 
https://huggingface.co/Qwen/Qwen3-Next-80B-A3B-Instruct?library=transformers
Available in transformers>=4.57.0

## Testing Done

- Hardware Type: <BLANK>
- [ ] run `make test` to ensure correctness
- [ ] run `make checkstyle` to ensure code style
- [ ] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by Tcc0403** [comment]
## Monkey patch test

```python
❯ python3 -m pytest test/transformers/test_monkey_patch.py -k qwen3_next
/home/tcc/Liger-Kernel/.venv/lib/python3.10/site-packages/pytest_asyncio/plugin.py:252: PytestDeprecationWarning: The configuration option "asyncio_default_fixture_loop_scope" is unset.
The event loop scope for asynchronous fixtures will default to the fixture caching scope. Future versions of pytest-asyncio will default the loop scope for asynchronous fixtures to function scope. Set the default fixture loop scope explicitly in order to avoid unexpected behavior in the future. Valid fixture loop scopes are: "function", "class", "module", "package", "session"

  warnings.warn(PytestDeprecationWarning(_DEFAULT_FIXTURE_LOOP_SCOPE_UNSET))
============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.12, pytest-8.3.4, pluggy-1.5.0
rootdir: /home/tcc/Liger-Kernel
configfile: pyproject.toml
plugins: rerunfailures-15.0, asyncio-1.2.0, xdist-3.6.1, cov-7.0.0, anyio-4.9.0
asyncio: mode=auto, debug=False, asyncio_default_fixture_loop_scope=None, asyncio_default_test_loop_scope=function
collected 39 items / 38 deselected / 1 selected

test/transformers/test_monkey_patch.py::test_apply_liger_kernel_to_instance_for_qwen3_next
------------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------------
WARNING  fla.utils:utils.py:56 Current Python version 3.10 is below the recommended 3.11 version. It is recommended to upgrade to Python 3.11 or higher for the best experience.
INFO     liger_kernel.transformers.monkey_patch:monkey_patch.py:2367 Applying Liger kernels to model instance with model type: qwen3_next with kwargs: {}
PASSED                                                                                                                                                                                                     [100%]
```
## Convergence test 

**(bf16 only since flash-linear-attention doesn't support fp32 input)**

without flash-linear-attention installed
```python
❯ python3 -m pytest test/convergence/bf16/* -k qwen3_next
/home/tcc/Liger-Kernel/.venv/lib/python3.10/site-packages/pytest_asyncio/plugin.py:252: PytestDeprecationWarning: The configuration option "asyncio_default_fixture_loop_scope" is unset.
The event loop scope for asynchronous fixtures will default to the fixture caching scope. Future versions of pytest-asyncio will default the loop scope for asynchronous fixtures to function scope. Set the default fixture loop scope explicitly in order to avoid unexpected behavior in the future. Valid fixture loop scopes are: "function", "class", "module", "package", "session"

  warnings.warn(PytestDeprecationWarning(_DEFAULT_FIXTURE_LOOP_SCOPE_UNSET))
============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.12, pytest-8.3.4, pluggy-1.5.0
rootdir: /home/tcc/Liger-Kernel
configfile: pyproject.toml
plugins: rerunfailures-15.0, asyncio-1.2.0, xdist-3.6.1, cov-7.0.0, anyio-4.9.0
asyncio: mode=auto, debug=False, asyncio_default_fixture_loop_scope=None, asyncio_default_test_loop_scope=function
collected 55 items / 53 deselected / 2 selected

test/convergence/bf16/test_mini_models.py::test_mini_model[mini_qwen3_next-32-1e-05-dtype22-0.01-0.01-0.1-0.1-0.01-0.01] PASSED                                                                            [ 50%]
test/convergence/bf16/test_mini_models_with_logits.py::test_mini_model[mini_qwen3_next-32-1e-05-dtype22-0.01-0.01-0.1-0.1-0.01-0.01] PASSED                                                                [100%]
```
with flash-linear-attention installed
```python
❯ uv pip install flash-linear-attention
Resolved 72 packages in 734ms
Installed 2 packages in 48ms
 + fla-core==0.3.2
 + flash-linear-attention==0.3.2
❯ python3 -m pytest test/convergence/bf16/* -k qwen3_next
/home/tcc/Liger-Kernel/.venv/lib/python3.10/site-packages/pytest_asyncio/plugin.py:252: PytestDeprecationWarning: The configuration option "asyncio_default_fixture_loop_scope" is unset.
The event loop scope for asynchronous fixtures will default to the fixture caching scope. Future versions of pytest-asyncio will default the loop scope for asynchronous fixtures to function scope. Set the default fixture loop scope explicitly in order to avoid unexpected behavior in the future. Valid fixture loop scopes are: "function", "class", "module", "package", "session"

  warnings.warn(PytestDeprecationWarning(_DEFAULT_FIXTURE_LOOP_SCOPE_UNSET))
============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.12, pytest-8.3.4, pluggy-1.5.0
rootdir: /home/tcc/Liger-Kernel
configfile: pyproject.toml
plugins: rerunfailures-15.0, asyncio-1.2.0, xdist-3.6.1, cov-7.0.0, anyio-4.9.0
asyncio: mode=auto, debug=False, asyncio_default_fixture_loop_scope=None, asyncio_default_test_loop_scope=function
collecting ...
---------------------------------------------------------------------------------------------- live log collection -----------------------------------------------------------------------------------------------
WARNING  fla.utils:utils.py:56 Current Python version 3.10 is below the recommended 3.11 version. It is recommended to upgrade to Python 3.11 or higher for the best experience.
collected 55 items / 53 deselected / 2 selected

test/convergence/bf16/test_mini_models.py::test_mini_model[mini_qwen3_next-32-1e-05-dtype22-0.01-0.01-0.1-0.1-0.01-0.01] PASSED                                                                            [ 50%]
test/convergence/bf16/test_mini_models_with_logits.py::test_mini_model[mini_qwen3_next-32-1e-05-dtype22-0.01-0.01-0.1-0.1-0.01-0.01] PASSED                                                                [100%]
```

**Comment by Tcc0403** [comment]
@michaelroyzen Qwen3Next support should be all done with this branch. RoPE patch currently has nan issue, requiring further investigation. But other modules should work and are compatible with flash-linear-attention package. Feel free to ping me if you got any issue!

**Review by Tcc0403 — APPROVED** [review]

**Comment by michaelroyzen** [comment]
Thank you @Tcc0403 @vvvdwbvvv, running my own correctness tests now!

Update: it works as expected, the loss and gradient norm values match up. Great work.
